### ISSUE TITLE
Result file of classification

### ISSUE THREAD

**Issue by narges-tk** [yanx27/Pointnet_Pointnet2_pytorch#247]
Hi!
I managed to run the code but I found no result file, ideally a file with point clouds and the label of classes. Could you please help me in this regard?

**Comment by Lobbyism** [comment]
Same issue. Are we just expected to add changes to serialize `pred_choice`?... Not sure why it's already done, but I'll do that.

![Image](https://github.com/user-attachments/assets/061cf1a8-82e6-4d89-a39f-2975c2782641)

**Comment by Lobbyism** [comment]
```python
def test(model, loader, num_class=40, vote_num=1):
    mean_correct = []
    classifier = model.eval()
    class_acc = np.zeros((num_class, 3))
    all_predictions = []

    for j, (points, target) in tqdm(enumerate(loader), total=len(loader)):
        if not args.use_cpu:
            points, target = points.cuda(), target.cuda()

        points = points.transpose(2, 1)
        vote_pool = torch.zeros(target.size()[0], num_class).cuda()

        for _ in range(vote_num):
            pred, _ = classifier(points)
            vote_pool += pred
        pred = vote_pool / vote_num
        pred_choice = pred.data.max(1)[1]
        all_predictions.append(pred_choice.cpu().numpy())

        for cat in np.unique(target.cpu()):
            classacc = pred_choice[target == cat].eq(target[target == cat].long().data).cpu().sum()
            class_acc[cat, 0] += classacc.item() / float(points[target == cat].size()[0])
            class_acc[cat, 1] += 1
        correct = pred_choice.eq(target.long().data).cpu().sum()
        mean_correct.append(correct.item() / float(points.size()[0]))

    class_acc[:, 2] = class_acc[:, 0] / class_acc[:, 1]
    class_acc = np.mean(class_acc[:, 2])
    instance_acc = np.mean(mean_correct)
    all_predictions = np.concatenate(all_predictions, axis=0)
    return instance_acc, class_acc, all_predictions
```
